src/settings.cpp

```diff
@@ -1879,6 +1886,40 @@ void DeleteGRFPresetFromConfig(const char *config_name)
 	delete ini;
 }
 
+/**
+ * Handle setting a value from a command, e.g. CMD_CHANGE_SETTING or CMD_CHANGE_COMPANY_SETTING.
+ * This performs validation of the input value and the appropriate callbacks as well as saving if needed.
+ * @param object The object the setting is in.
+ * @param newval The new value for the setting.
+ */
+void IntSettingDesc::SetFromCommand(const void *object, int32 newval) const
```

Comment:
Maybe this could use a better name and not be tight to commands. Mainly as this PR itself already violates that, as the console 'set' command also sends up here, if I understand this correctly.

Isn't this just `ChangeValue` or something?